src/box/box.cc

```diff
@@ -2605,6 +2616,12 @@ relay_get_sync_on_start(struct trigger *trigger, void *event)
 	/* Already accounted. */
 	if (bit_test(&data->collected_vclock_map, id))
 		return 0;
+
+	++data->ref_count;
```

Comment:
Let's introduce helpers `sync_trigger_data_ref` and `sync_trigger_data_unref`
which would increment and decrement the `ref_count`, and free the data, if needed